### PR TITLE
Fix return wrapping in extension

### PR DESCRIPTION
**Pull request by m-nash**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Generation/MgmtExtensionWriter.cs [hunk 1]
```diff
@@ -59,7 +59,7 @@ public MgmtExtensionWriter(CodeWriter writer, MgmtExtensions extensions, BuildCo
 
         protected Parameter ExtensionParameter { get; }
 
-        protected void WriteMethodWrapper(MgmtClientOperation clientOperation, bool async)
+        protected void WriteMethodWrapper(MgmtClientOperation clientOperation, bool isAsync)
```
Comment (annelo-msft) [inline src/AutoRest.CSharp/Mgmt/Generation/MgmtExtensionWriter.cs:62]: Nit/Curious: why this change?
We have a convention elsewhere to use async, wbn to be consistent: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/BearerTokenAuthenticationPolicy.cs#L113

Comment (m-nash) [reply src/AutoRest.CSharp/Mgmt/Generation/MgmtExtensionWriter.cs:62]: bool should be verbs that ask a question IMHO, I do see a mixed style throughout the code

**Review by annelo-msft — APPROVED** [review]
The output files look to be consistent with the Azure SDK guidelines.  I haven't reviewed anything under src/
